### PR TITLE
Split coqc/coqdep rules into separate functions

### PR DESCRIPTION
**Pull request by rgrinberg**
This refactoring will make it easier to understand that the extraction stanza
will introduce in an upcoming PR.

### CODE REVIEW

**Comment by rgrinberg** [comment]
@ejgallego ping?

File: src/dune/coq_rules.ml [hunk 1]
```diff
@@ -121,44 +128,60 @@ let setup_rule ~expander ~dir ~cc ~source_rule ~coq_flags ~file_flags
...
+  let open Build.With_targets.O in
   (* coqdep needs the full source + plugin's mlpack to be present :( *)
-  let coqdep_rule =
-    (* This is weird stuff in order to adapt the rule so we can reuse ml_iflags
-       :( I wish we had more flexible typing. *)
-    Build.with_no_targets mlpack_rule
-    >>> Build.with_no_targets source_rule
-    >>> Command.run ~dir ~stdout_to cc.coqdep cd_arg
-  in
+  (* This is weird stuff in order to adapt the rule so we can reuse ml_iflags :(
```
Comment (ejgallego) [inline src/dune/coq_rules.ml]: This comment is outdated [refactorings in build/args made it obsolete]

**Comment by ejgallego** [comment]
Not merging in case you feel like fixing the comment.

**Comment by rgrinberg** [comment]
> I've already pushed a few fixes to Coq upstream so boot mode has common flag handling for coqdep / coqc, that should allow for further cleanups [will submit soon]

Looking forward to see this!

> still not clear the rule scheme for coqdep, at least for version 1.0 of the coq lang, having a kind of -modules output is hard due to coqdep having to resolve paths in qualified imports. This PR would greatly help if we would like to implement a "one coqdep call per theory"

Is there a coq ticket to track this issue?

**Comment by ejgallego** [comment]
> Is there a Coq ticket to track this issue?

If you mean about `-modules` I'm not sure now, there is really a lot of issues related to lib organization; we need to do some large cleanup first, but definitively was discussed; I guess we will centralize this kind of discussions in the 2.0 library model issue.

For now, the need to resolve `From Foo Require Bar` to `Foo.**.Bar` limits us quite a bit :/

Regarding generating a `theory.d` file vs individual `file.v.d` files IMHO that's orthogonal to Coq but our choice; the way things are setup specially after #2053 makes this easy.

Whether that's worth it, I dunno, but in some scenarios we are doing a few thousand `coqdep` calls and setup is expensive. On the other this is more of a micro-optimization for me so I'm not in a rush; just wanted to mention the possibility as you are tweaking coqdep rules.

**Pull request merged**